src/data/mystery-encounters/mystery-encounter.ts

```diff
@@ -0,0 +1,963 @@
+   */
+  skipEnemyBattleTurns: boolean;
+  /**
+   * If true, will skip COMMAND input and go straight to FIGHT (move select) input menu
+   */
+  skipToFightInput: boolean;
+
+  /**
+   * Event callback functions
+   */
+  /** Event when Encounter is first loaded, use it for data conditioning */
+  onInit?: (scene: BattleScene) => boolean;
+  /** Event when battlefield visuals have finished sliding in and the encounter dialogue begins */
+  onVisualsStart?: (scene: BattleScene) => boolean;
+  /** Event triggered prior to {@linkcode CommandPhase}, during {@linkcode TurnInitPhase} */
+  onTurnStart?: (scene: BattleScene) => boolean;
+  /** Event prior to any rewards logic in {@linkcode MysteryEncounterRewardsPhase} */
+  onRewards?: (scene: BattleScene) => Promise<void>;
+  /** Will provide the player party EXP before rewards are displayed for that wave */
+  doEncounterExp?: (scene: BattleScene) => boolean;
+  /** Will provide the player a rewards shop for that wave */
+  doEncounterRewards?: (scene: BattleScene) => boolean;
+  /** Will execute callback during VictoryPhase of a continuousEncounter */
+  doContinueEncounter?: (scene: BattleScene) => Promise<void>;
+
+  /**
+   * Requirements
+   */
+  requirements: EncounterSceneRequirement[];
+  /** Primary Pokemon is a single pokemon randomly selected from the party that meet ALL primary pokemon requirements */
+  primaryPokemonRequirements: EncounterPokemonRequirement[];
+  /**
+   * Secondary Pokemon are pokemon that meet ALL secondary pokemon requirements
+   * Note that an individual requirement may require multiple pokemon, but the resulting pokemon after all secondary requirements are met may be lower than expected
+   * If the primary pokemon and secondary pokemon are the same and ExcludePrimaryFromSupportRequirements flag is true, primary pokemon may be promoted from secondary pool
+   */
+  secondaryPokemonRequirements: EncounterPokemonRequirement[];
+  excludePrimaryFromSupportRequirements: boolean;
+  primaryPokemon?: PlayerPokemon;
+  secondaryPokemon?: PlayerPokemon[];
+
+  /**
+   * Post-construct / Auto-populated params
+   */
+
+  /**
+   * Dialogue object containing all the dialogue, messages, tooltips, etc. for an encounter
+   */
+  dialogue: MysteryEncounterDialogue;
+  /**
+   * Data used for setting up/initializing enemy party in battles
+   * Can store multiple configs so that one can be chosen based on option selected
+   * Should usually be defined in `onInit()` or `onPreOptionPhase()`
+   */
+  enemyPartyConfigs: EnemyPartyConfig[];
+  /**
+   * Object instance containing sprite data for an encounter when it is being spawned
+   * Otherwise, will be undefined
+   * You probably shouldn't do anything directly with this unless you have a very specific need
+   */
+  introVisuals?: MysteryEncounterIntroVisuals;
+
+  /**
+   * Flags
+   */
```

Comment:
you could use 
```suggestion
  // #region Flags
```

Make sure to close it with
```ts
  // #endregion
```

this will show up like this in vs-codium minimap:
<img width="154" alt="image" src="https://github.com/user-attachments/assets/1cc24b61-200f-4d17-8d84-08bff26c1baa">